tests/e2e/ctl_v3_kv_test.go

```diff
@@ -216,6 +220,114 @@ func getRevTest(cx ctlCtx) {
 	}
 }
 
+func getMaxCreateRevTest(cx ctlCtx) {
+	var (
+		kvs = []kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}}
+	)
+	for i := range kvs {
+		if err := ctlV3Put(cx, kvs[i].key, kvs[i].val, ""); err != nil {
+			cx.t.Fatalf("getRevTest #%d: ctlV3Put error (%v)", i, err)
+		}
+	}
+
+	tests := []struct {
+		args []string
+
+		wkv []kv
+	}{
+		{[]string{"key", "--prefix", "--max-create-rev", "3"}, kvs[:2]},
+	}
+
+	for i, tt := range tests {
+		if err := ctlV3Get(cx, tt.args, tt.wkv...); err != nil {
+			cx.t.Errorf("getMaxCreateRevTest #%d: ctlV3Get error (%v)", i, err)
+		}
+	}
+}
+
+func getMinCreateRevTest(cx ctlCtx) {
+	var (
+		kvs = []kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}}
+	)
+	for i := range kvs {
+		if err := ctlV3Put(cx, kvs[i].key, kvs[i].val, ""); err != nil {
+			cx.t.Fatalf("getMinCreateRevTest #%d: ctlV3Put error (%v)", i, err)
+		}
+	}
+
+	tests := []struct {
+		args []string
+
+		wkv []kv
+	}{
+		{[]string{"key", "--prefix", "--min-create-rev", "3"}, kvs[1:]},
+	}
+
+	for i, tt := range tests {
+		if err := ctlV3Get(cx, tt.args, tt.wkv...); err != nil {
+			cx.t.Errorf("getTest #%d: ctlV3Get error (%v)", i, err)
+		}
+	}
+}
+
+func getMaxModRevTest(cx ctlCtx) {
+	var (
+		kvs = []kv{ //                      store revision | key create revision | key modify revision
+			{"key1", "val1"}, //         2                   2                   2
+			{"key2", "val2"}, //         3                   3                   3
+			{"key2", "val3"}, //         4                   2                   4
+		}
+	)
+	for i := range kvs {
+		if err := ctlV3Put(cx, kvs[i].key, kvs[i].val, ""); err != nil {
+			cx.t.Fatalf("getRevTest #%d: ctlV3Put error (%v)", i, err)
+		}
+	}
+
+	tests := []struct {
+		args []string
+
+		wkv []kv
+	}{
+		{[]string{"key", "--prefix", "--max-mod-rev", "3"}, kvs[:1]},
+	}
+
+	for i, tt := range tests {
+		if err := ctlV3Get(cx, tt.args, tt.wkv...); err != nil {
+			cx.t.Errorf("getMaxModRevTest #%d: ctlV3Get error (%v)", i, err)
+		}
+	}
+}
+
+func getMinModRevTest(cx ctlCtx) {
+	var (
+		kvs = []kv{ //                      store revision | key create revision | key modify revision
+			{"key1", "val1"}, //         2                   2                   2
+			{"key2", "val2"}, //         3                   3                   3
+			{"key1", "val3"}, //         4                   2                   4
+		}
+	)
+	for i := range kvs {
+		if err := ctlV3Put(cx, kvs[i].key, kvs[i].val, ""); err != nil {
+			cx.t.Fatalf("getRevTest #%d: ctlV3Put error (%v)", i, err)
+		}
+	}
+
+	tests := []struct {
+		args []string
+
+		wkv []kv
+	}{
+		{[]string{"key", "--prefix", "--min-mod-rev", "4"}, kvs[2:]},
+	}
+
+	for i, tt := range tests {
+		if err := ctlV3Get(cx, tt.args, tt.wkv...); err != nil {
+			cx.t.Errorf("getMinModRevTest #%d: ctlV3Get error (%v)", i, err)
+		}
+	}
+}
+
```

Comment:
These functions have very similar implementations. I think we can just create one table-driven test case.

Reply:
Done.